rivescript/parser.py

```diff
@@ -535,13 +535,13 @@ def check_syntax(self, cmd, line):
             if parts[0] == "begin" and len(parts) > 1:
                 return "The 'begin' label takes no additional arguments, should be verbatim '> begin'"
             elif parts[0] == "topic":
-                match = re.match(RE.name_syntax, line)
-                if match:
+                search = re.search(RE.name_syntax, line)
+                if search:
                     return "Topics should be lowercased and contain only numbers and letters"
             elif parts[0] == "object":
-                match = re.match(RE.name_syntax, line)
-                if match:
-                    return "Objects can only contain numbers and letters"
+                search = re.search(RE.name_syntax, line)
+                if search:
+                    return "Objects can only contain numbers and letters"  # TODO Acceptance of uppercase letter?
```

Comment:
It seems safe enough to me to allow uppercase letters in object names. You may need to add a new syntax regexp to `regexp.py` to support that, like `r'[^A-Za-z0-9_\-\s]'`